pkg/reconciler/kubernetes/tektonpipelinesascode/tektonpipelinesascode.go

```diff
@@ -0,0 +1,274 @@
+
+// Check that our Reconciler implements controller.Reconciler
+var _ tpacreconciler.Interface = (*Reconciler)(nil)
+var _ tpacreconciler.Finalizer = (*Reconciler)(nil)
+
+// FinalizeKind removes all resources after deletion of a TektonTriggers.
+func (r *Reconciler) FinalizeKind(ctx context.Context, tpac *v1alpha1.TektonPipelinesAsCode) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+
+	err := r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().Delete(ctx, tpac.Status.InstallerSet, metav1.DeleteOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	if err := r.extension.Finalize(ctx, tpac); err != nil {
+		logger.Error("Failed to finalize platform resources", err)
+	}
+
+	return nil
+}
+
+// ReconcileKind compares the actual state with the desired, and attempts to
+// converge the two.
+func (r *Reconciler) ReconcileKind(ctx context.Context, tpac *v1alpha1.TektonPipelinesAsCode) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+	tpac.Status.InitializeConditions()
+
+	if tpac.GetName() != v1alpha1.PACResourceName {
+		msg := fmt.Sprintf("Resource ignored, Expected Name: %s, Got Name: %s",
+			v1alpha1.PACResourceName,
+			tpac.GetName(),
+		)
+		logger.Error(msg)
+		tpac.Status.MarkNotReady(msg)
+		return nil
+	}
+
+	// Make sure TektonPipeline & TektonTrigger is installed before proceeding with
+	// TektonPipelinesAsCode
+
+	if _, err := common.PipelineReady(r.pipelineInformer); err != nil {
+		if err.Error() == common.PipelineNotReady {
+			tpac.Status.MarkDependencyInstalling("tekton-pipelines is still installing")
+			// wait for pipeline status to change
+			return fmt.Errorf(common.PipelineNotReady)
+		}
+		// (tektonpipeline.operator.tekton.dev instance not available yet)
+		tpac.Status.MarkDependencyMissing("tekton-pipelines does not exist")
+		return err
+	}
+
+	if _, err := common.TriggerReady(r.triggerInformer); err != nil {
+		if err.Error() == common.TriggerNotReady {
+			tpac.Status.MarkDependencyInstalling("tekton-triggers is still installing")
+			// wait for trigger status to change
+			return fmt.Errorf(common.TriggerNotReady)
+		}
+		// (tektontrigger.operator.tekton.dev instance not available yet)
+		tpac.Status.MarkDependencyMissing("tekton-triggers does not exist")
+		return err
+	}
+
+	tpac.Status.MarkDependenciesInstalled()
+
+	// Pass the object through defaulting
+	tpac.SetDefaults(ctx)
+
+	if err := r.extension.PreReconcile(ctx, tpac); err != nil {
+		tpac.Status.MarkPreReconcilerFailed(err.Error())
+		return err
+	}
+
+	tpac.Status.MarkPreReconcilerComplete()
+
+	exist, err := checkIfInstallerSetExist(ctx, r.operatorClientSet, r.version, tpac)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+
+		manifest := r.manifest
+		if err := r.transform(ctx, &manifest, tpac); err != nil {
+			logger.Error("manifest transformation failed:  ", err)
+			return err
+		}
+
+		return createInstallerSet(ctx, r.operatorClientSet, tpac, r.manifest, r.version, "pac")
+	}
+
```

Comment:
don't we need the upgrade scenario as well. ie, `installerset` exist but the releaseVersion has changed

Reply:
oh, i see, https://github.com/tektoncd/operator/pull/550/files#r766263482

Reply:
but, how will the targetNamespace changed (or any spec) scenario be handled.

Reply:
updated !